core/src/main/java/org/elasticsearch/search/aggregations/bucket/significant/InternalMappedSignificantTerms.java

```diff
@@ -123,4 +124,19 @@ protected boolean doEquals(Object obj) {
     protected int doHashCode() {
         return Objects.hash(super.doHashCode(), format, subsetSize, supersetSize, significanceHeuristic, buckets, bucketMap);
     }
+
+    @Override
+    public XContentBuilder doXContentBody(XContentBuilder builder, Params params) throws IOException {
+        builder.field(CommonFields.DOC_COUNT.getPreferredName(), subsetSize);
+        builder.startArray(CommonFields.BUCKETS.getPreferredName());
+        for (Bucket bucket : buckets) {
+            //There is a condition (presumably when only one shard has a bucket?) where reduce is not called
+            // and I end up with buckets that contravene the user's min_doc_count criteria in my reducer
+            if (bucket.subsetDf >= minDocCount) {
```

Comment:
Ooop sorry, I wanted to add a comment about that and ping Mark. 

Tests passed and I'm curious to know why there's a different behavior between String & Long here. @markharwood do you have any idea?